core/src/main/java/org/neo4j/ogm/context/MappingContext.java

```diff
@@ -64,7 +64,7 @@ public Object addNodeEntity(Object entity, Long id) {
             entity = nodeEntityRegister.get(id);
             addType(entity.getClass(), entity, id);
         }
-        remember(entity);
+        remember(entity, false);
         return entity;
```

Comment:
The use of false here seems un-intuitive, in that we never update set the hash - even for new objects. Is this because we rely on the id being null in these cases in order to detect it is dirty? In any case I think it needs some tests around it.